mlbench_core/models/tensorflow/resnet_run_loop.py

```diff
@@ -0,0 +1,388 @@
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import math
+import multiprocessing
+import os
+
+from absl import flags
+import tensorflow as tf
+from tensorflow.contrib.data.python.ops import threadpool
+
+from . import resnet_model
+
+from mlbench_core.utils.tensorflow import distribution_utils
+
+
+def resnet_model_fn(features, labels, mode, model_class,
+                    resnet_size, weight_decay, learning_rate_fn, momentum,
+                    data_format, resnet_version, loss_scale,
+                    loss_filter_fn=None, dtype=resnet_model.DEFAULT_DTYPE,
+                    fine_tune=False):
+    """Shared functionality for different resnet model_fns.
+    Initializes the ResnetModel representing the model layers
+    and uses that model to build the necessary EstimatorSpecs for
+    the `mode` in question. For training, this means building losses,
+    the optimizer, and the train op that get passed into the EstimatorSpec.
+    For evaluation and prediction, the EstimatorSpec is returned without
+    a train op, but with the necessary parameters for the given mode.
+    Args:
+        features: tensor representing input images
+        labels: tensor representing class labels for all input images
+        mode: current estimator mode; should be one of
+        `tf.estimator.ModeKeys.TRAIN`, `EVALUATE`, `PREDICT`
+        model_class: a class representing a TensorFlow model that has a __call__
+        function. We assume here that this is a subclass of ResnetModel.
+        resnet_size: A single integer for the size of the ResNet model.
+        weight_decay: weight decay loss rate used to regularize learned variables.
+        learning_rate_fn: function that returns the current learning rate given
+        the current global_step
+        momentum: momentum term used for optimization
+        data_format: Input format ('channels_last', 'channels_first', or None).
+        If set to None, the format is dependent on whether a GPU is available.
+        resnet_version: Integer representing which version of the ResNet network to
+        use. See README for details. Valid values: [1, 2]
+        loss_scale: The factor to scale the loss for numerical stability. A detailed
+        summary is present in the arg parser help text.
+        loss_filter_fn: function that takes a string variable name and returns
+        True if the var should be included in loss calculation, and False
+        otherwise. If None, batch_normalization variables will be excluded
+        from the loss.
+        dtype: the TensorFlow dtype to use for calculations.
+        fine_tune: If True only train the dense layers(final layers).
+    Returns:
+        EstimatorSpec parameterized according to the input params and the
+        current mode.
+    """
+
+    # Generate a summary node for the images
+    tf.summary.image('images', features, max_outputs=6)
```

Comment:
tf.summary is only used for showing things in e.g. tensorboard, right? Do we need it on our setting?